pkg/cloud/devicemanager/manager.go

```diff
@@ -119,6 +119,10 @@ func (d *deviceManager) NewDevice(instance *ec2.Instance, volumeID string) (*Dev
 
 	// Check if this volume is already assigned a device on this machine
 	if path := d.getPath(inUse, volumeID); path != "" {
+		if status := d.getDeviceStatus(volumeID, instance); status == "detaching" {
```

Comment:
nit: Can we rename this field to volumeStatus?

Reply:
Can you also put a comment here that this fix will only handle the detaching inconsistency, if the status change from detach to attached it will still fail (Just for tracking purpose)